src/reductiveContext.re

```diff
@@ -0,0 +1,91 @@
+[@bs.config {jsx: 3}];
+module type Config = {
+  type state;
+  type action;
+
+  let store: Reductive.Store.t(action, state);
+};
+
+module Make = (Config: Config) => {
+  let storeContext = React.createContext(Config.store);
+
+  module ContextProvider = {
+    let make = React.Context.provider(storeContext);
+    let makeProps = (~value, ~children, ()) => {
+      "value": value,
+      "children": children,
+    };
+  };
+
+  module Provider = {
+    [@react.component]
+    let make = (~children) => {
+      <ContextProvider value=Config.store> children </ContextProvider>;
+    };
+  };
+
+  let useSelector = selector => {
+    let storeFromContext = React.useContext(storeContext);
+    let (_, forceRerender) = React.useReducer((s, _) => s + 1, 0);
+
+    // initialize ref to None to avoid reevaluating selector function on each rerender
+    let latestSelectedState = React.useRef(None);
+    let latestSelector = React.useRef(selector);
+
+    // selector function might change if using components' props to select state
+    React.useLayoutEffect1(
```

Comment:
We should really try to avoid using `useLayoutEffect` if we can. It's not really intended for this kind of general state management usecase.

Reply:
The new implementation based on `useSubscription` is using `useEffect` instead.